### ISSUE TITLE
Possible to add tags with spaces when TAG_SPACES_ALLOWED = False

### ISSUE THREAD

**Issue by m1kola** [wagtail/wagtail#4109]
### Issue Summary

It's possible to add spaces into tags with `TAG_SPACES_ALLOWED = False`.

### Steps to Reproduce

0. Set `TAG_SPACES_ALLOWED = False` in your settings
1. Open the editing interface for a document or a page with the default Wagtail tags input
2. Copy few words with a space inside into clipboard
3. Paste into the tag input
3. Hit the save/publish button

**Result:** Tag with a space will be saved into a database

We need to prevent copy-pasting when `TAG_SPACES_ALLOWED is False` on front-end and/or implement validation on back-end.

### Technical details

* Wagtail version: Wagtail 1.13.1
* Browser version: 62.0.3202.94 (64-bit, MacOS 10.13.1)

**Comment by gasman** [comment]
This would be the responsibility of the `tag-it.js` library, and it looks like one of these PRs might implement this behaviour: https://github.com/aehlke/tag-it/pull/95, https://github.com/aehlke/tag-it/pull/261

(The tag-it github project looks somewhat unmaintained, and IIRC we're also using a not-quite-up-to-date version of it)

**Comment by m1kola** [comment]
> This would be the responsibility of the `tag-it.js` library

Do you think we don't need back-end validation for tags?

**Comment by gasman** [comment]
@m1kola No, I don't think so. TAG_SPACES_ALLOWED was never really intended to _enforce_ a ban on spaces in tags - even with `TAG_SPACES_ALLOWED = False`, it's intentional that users should be able to insert a tag with a space in it by double-quoting it.

The name is perhaps a bit misleading, but it's basically a consequence of the "holy war" (see https://github.com/wagtail/wagtail/pull/2207#issuecomment-196354247 as an entry point...) about how multi-word tags should be written, and thus which behaviour we should optimise for in the UX design:

* `TAG_SPACES_ALLOWED = False`: the "oldschool hacker" viewpoint. Tags should generally be single tokens, and multi-word ones should be `joined-with-hyphens`. This frees up the space character to be the delimiter between tags, which makes it nice and fluid to type in a long list of tags quickly. If you _really_ want a tag with a space in it, enclose it in quotes.
* `TAG_SPACES_ALLOWED = True`: the "let me write like a human dammit" viewpoint. Putting spaces in tags is a perfectly normal thing to do and shouldn't require any weird quoting tricks; tags should be separated with a comma.

**Comment by m1kola** [comment]
> No, I don't think so. TAG_SPACES_ALLOWED was never really intended to _enforce_ a ban on spaces in tags - even with `TAG_SPACES_ALLOWED = False`

Then we need to update [documentation](http://docs.wagtail.io/en/v1.13.1/advanced_topics/settings.html#multi-word-tags). Currently it says the following:

> ```python
> TAG_SPACES_ALLOWED = False
> ```
> Tags can only consist of a single word, no spaces allowed. The default setting is `True` (spaces in tags are allowed).

I interpreted it as "with `TAG_SPACES_ALLOWED = False` I can't use spaces in tags". The part with "no spaces allowed" sounds very convincing :)

**Comment by brylie** [comment]
I'm not sure if this is the right issue, but we are actually having trouble inserting tags with spaces. When we add double quotes around the tag, it still splits it into two tags -- each having a quote character.

**Comment by RoseOfSteel** [comment]
Another tag/validation issue?

You can add a tag with commas in the snippet name. But then on the tagged page, it shows up as multiple tags.

To reproduce:
1. Add a tag with one name "Budgets" to a page. Save the page.
2. Go to Snippet > Tags
3. Change the "Budgets" name to Bills, Budgets, & Shopping. I've tried with and without quotes. Ref: First screenshot below.
4. Look at the list of tags. Everything should look fine.
5. The page now has three tags: "Bills", "Budgets", "&Shopping". Ref: Second screenshot below.
6. You can either save the page or click out of it.
7. If you go back to Snippet > Tags, now you see the original invalid tag plus the multiple tags that were created.

It seems like you shouldn't be able to name the tag something if it's invalid.

Based on the comment above about spaces ("TAG_SPACES_ALLOWED was never really intended to enforce a ban on spaces in tags"), I'm guessing this is a similar situation where the tagit library isn't necessarily enforcing the tag format? Or is this expected behavior?

![wagtail_tag_error_2](https://github.com/wagtail/wagtail/assets/149306708/7bbbfbd2-4352-4327-b9fe-e46bef1de337)
![wagtail_tag_error_1](https://github.com/wagtail/wagtail/assets/149306708/f55d7c39-7f28-4019-8fbd-82c699051e49)
